DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/CourseSetupController.cs

```diff
@@ -130,34 +126,59 @@ public IActionResult AddCourseNew()
 
         [ServiceFilter(typeof(RedirectEmptySessionData<AddNewCentreCourseData>))]
         [HttpGet("AddCourse/SelectCourse")]
-        public IActionResult SelectCourse(int? topicId = null)
+        public IActionResult SelectCourse(
+            string? categoryFilterBy = null,
+            string? topicFilterBy = null
+        )
         {
-            var data = TempData.Peek<AddNewCentreCourseData>()!;
-
-            var model = GetSelectCourseViewModel(data!.Application?.ApplicationId, topicId);
+            var model = GetSelectCourseViewModel(categoryFilterBy, topicFilterBy);
 
             return View("AddNewCentreCourse/SelectCourse", model);
         }
 
+        [Route("AddCourse/SelectCourseAllCourses")]
+        public IActionResult SelectCourseAllCourses()
+        {
+            var centreId = User.GetCentreId();
+            var adminCategoryFilter = User.GetAdminCourseCategoryFilter();
+
+            var applications = courseService
+                .GetApplicationOptionsAlphabeticalListForCentre(centreId, adminCategoryFilter);
+            var categories = courseService.GetCategoriesForCentreAndCentrallyManagedCourses(centreId);
+            var topics = courseService.GetTopicsForCentreAndCentrallyManagedCourses(centreId);
+
+            var model = new SelectCourseAllCoursesViewModel(applications, categories, topics);
+            return View("AddNewCentreCourse/SelectCourseAllCourses", model);
+        }
+
         [ServiceFilter(typeof(RedirectEmptySessionData<AddNewCentreCourseData>))]
         [HttpPost("AddCourse/SelectCourse")]
-        public IActionResult SelectCourse(SelectCourseViewModel model)
+        public IActionResult SelectCourse(
+            int? applicationId,
+            string? categoryFilterBy = null,
+            string? topicFilterBy = null
+        )
         {
             var data = TempData.Peek<AddNewCentreCourseData>()!;
 
-            if (!ModelState.IsValid)
+            if (applicationId == null)
             {
+                ModelState.AddModelError("ApplicationId", "Select a course");

```

Comment:
Added the model error in the controller because the SelectCourseViewModel can't be posted since it inherits from BaseSearchablePageViewModel. If someone can think of a better way please shout!